### PR TITLE
Add settings option for "sfx_on_idle" which allows you to play a custom SFX alongside an Idle emote

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Makes https://github.com/AttorneyOnline/AO2-Client/pull/656 into a settings option

https://user-images.githubusercontent.com/3470436/181303514-94a85050-4782-4de0-9164-33f5c3f9715a.mp4

### CODE REVIEW

**Review by oldmud0 — DISMISSED** [review]
I had to read the old PR and the docs again to make sure this would accurately bring back pre-2.10 behavior. Seems to be the case

**Comment by in1tiate** [comment]
Doesn't this introduce possible confusion when some servers trigger different client behavior than others? The reason feature flags don't do this is because they're for *features,* and this isn't a feature so much as it is a pointless change in functionality.

**Comment by Crystalwarrior** [comment]
> Doesn't this introduce possible confusion when some servers trigger different client behavior than others? The reason feature flags don't do this is because they're for _features,_ and this isn't a feature so much as it is a pointless change in functionality.

Actually I might find a way to make this work without having a feature flag in the first place

**Comment by in1tiate** [comment]
Can confirm that it is extremely confusing behavior for sound effects to play even on idle. This absolutely needs to be something the server toggles on.

t. was insanely confused about this

**Comment by Crystalwarrior** [comment]
Overhauled this PR to resolve all the FL packet associated issues and make this a simple dumb settings option lol

**Comment by Salanto** [comment]
This has merge conflicts, lol

File: src/aooptionsdialog.cpp [hunk 1]
```diff
@@ -465,6 +465,18 @@ AOOptionsDialog::AOOptionsDialog(QWidget *parent, AOApplication *p_ao_app)
 
   ui_gameplay_form->setWidget(row, QFormLayout::FieldRole, ui_category_stop_cb);
 
+  row += 1;
+  ui_sfx_on_idle_lbl = new QLabel(ui_form_layout_widget);
+  ui_sfx_on_idle_lbl->setText(tr("Play SFX on Idle:"));
```
Comment (oldmud0) [inline src/aooptionsdialog.cpp:470]: misleading label. should be "Always Send SFX" since this has to do with client to server.

Comment (Crystalwarrior) [reply src/aooptionsdialog.cpp:470]: I agree

**Comment by stonedDiscord** [comment]
i hate this

**Review by stonedDiscord — APPROVED** [review]

**Pull request merged**